data/provider/src/main/java/tech/pegasys/teku/api/NodeDataProvider.java

```diff
@@ -89,11 +88,10 @@ public NodeDataProvider(
     this.payloadAttributesCalculator = payloadAttributesCalculator;
   }
 
-  public List<Attestation> getAttestations(
+  public List<tech.pegasys.teku.spec.datastructures.operations.Attestation> getAttestations(
       Optional<UInt64> maybeSlot, Optional<UInt64> maybeCommitteeIndex) {
     return attestationPool
         .getAttestations(maybeSlot, maybeCommitteeIndex)
-        .map(Attestation::new)
         .collect(Collectors.toList());
```

Comment:
we should just be able to return `attestationPool.getAttestations` now i think...

Reply:
Ok it's doing a bunch of filtering etc on the other end which is why its a stream, and it was probably left as a stream because we needed to map it anyway.
If we don't need to map it now, we can change it back to a list being returned (which will effect a couple of tests) and then its significantly cleaner over this side of the fence...